### PR TITLE
feat(DTFS2-6576): Updated `BSS/EWCS` eligibility criteria

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
`BSS/EWCS` eligibility criterion require threshold update in accordance with the latest MGA.

## Resolution :heavy_check_mark:
* Introduced `v7` for BSS/EWCS EC.
* Updated mocks, fixtures and mock-data.
* Updated TFM content string.

## Miscellaneous :heavy_plus_sign:
* Dependencies updates.
* Documentation update.
* Added missing BSS/EWCS EC `v6` to the mocked data.
* Updated google-github-actions/release-please-action to `v4`.

### CODE REVIEW

File: utils/mock-data-loader/bss/eligibilityCriteria.js [hunk 1]
```diff
@@ -266,4 +266,98 @@ module.exports = [
...
+      },
+      {
+        id: 12,
+        description:
+          'The period between the Cover Start Date and the Cover End Date does not exceed for a Bond, the Bond Maximum Cover Period; and for a Loan, the Loan Maximum Cover Period',
+      },
+      {
+        id: 13,
+        description:
+          'The Covered Bank Exposure under the Transaction converted (as at the date this representation is made) for this purpose into the Base Currency) is not more than the lesser of the Available Facility; and the Available Obligor Covered Exposure Limit',
```
Comment (lmarrai-sw) [inline utils/mock-data-loader/bss/eligibilityCriteria.js:335]:
```suggestion
          'The Covered Bank Exposure under the Transaction (converted (as at the date this representation is made) for this purpose into the Base Currency) is not more than the lesser of: the Available Facility; and the Available Obligor Covered Exposure Limit.',
```
I think we should:
- add the missing opening bracket before `converted`
- have a full stop at the end
- keep the colon before the parts of the text that are bullet points in the TFM version of the text

What do you think?
If we change it then we need to make the same change in all the mock data that has this EC as well

File: utils/mock-data-loader/bss/eligibilityCriteria.js [hunk 2]
```diff
@@ -266,4 +266,98 @@ module.exports = [
...
+    criteria: [
+      {
+        id: 11,
+        description:
+          'The Supplier has confirmed in its Supplier Declaration that the Supply Contract does not involve agents and the Bank is not aware that any of the information contained within it is inaccurate.',
+      },
+      {
+        id: 12,
+        description:
+          'The period between the Cover Start Date and the Cover End Date does not exceed for a Bond, the Bond Maximum Cover Period; and for a Loan, the Loan Maximum Cover Period',
```
Comment (lmarrai-sw) [inline utils/mock-data-loader/bss/eligibilityCriteria.js:330]:
```suggestion
          'The period between the Cover Start Date and the Cover End Date does not exceed: for a Bond, the Bond Maximum Cover Period; and for a Loan, the Loan Maximum Cover Period.',
```
I think we should:
- have a full stop at the end
- keep the colon before the parts of the text that are bullet points in the TFM version of the text

What do you think?
If we change it then we need to make the same change in all the mock data that has this EC as well

File: trade-finance-manager-api/src/v1/content-strings/index.js [hunk 3]
```diff
@@ -294,11 +312,8 @@ const DEAL = {
           textList: [],
         },
         18: {
-          text: 'The fees, interest rate and/or risk margin apply to the whole Cover Period, and have been set in accordance with:',
-          textList: [
-            'The bank’s normal pricing policies',
-            'Include, if any, pricing requirements set by UKEF',
-          ],
+          text: "Any applicable fee-s-, interest rate and/or Risk Margin Fee apply to the whole Cover Period of the Covered Transaction, and have been set in accordance with the Bank's normal pricing policies and include, if any, overall pricing requirements notified by UKEF.",
```
Comment (lmarrai-sw) [inline trade-finance-manager-api/src/v1/content-strings/index.js:315]: Is `fee-s-` intentional here, or is it a typo on the ACs?

**Review by lmarrai-sw — CHANGES_REQUESTED** [review]
This looks good thanks @abhi-markan, I've just commented on a few wording things

File: trade-finance-manager-api/src/v1/content-strings/index.js (hunk 3 above)
Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/content-strings/index.js:315]: Have asked, thank you for raising this.

File: utils/mock-data-loader/bss/eligibilityCriteria.js (hunk 2 above)
Comment (abhi-markan) [reply utils/mock-data-loader/bss/eligibilityCriteria.js:330]: Have updated.

File: utils/mock-data-loader/gef/eligibilityCriteria.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ const ELIGIBILITY_CRITERIA = [
       {
         id: 12,
         name: 'coverStart',
-        text: 'The period between the Cover Start Date and the Cover End Date does not exceed the Facility Maximum Cover Period.',
+        text: 'The period between the Cover Start Date and the Cover End Date does not exceed: the Facility Maximum Cover Period.',
```
Comment (lmarrai-sw) [inline utils/mock-data-loader/gef/eligibilityCriteria.js:11]: @abhi-markan I think we've accidentally changed the GEF ECs as well (here and in some of the test deals/ECs)

File: trade-finance-manager-api/src/v1/content-strings/index.js [hunk 5]
```diff
@@ -272,11 +272,11 @@ const DEAL = {
           textList: [],
         },
         12: {
-          text: 'The period between the Cover Start Date and the Cover End Date does not exceed:',
+          text: 'The period between the Cover Start Date and the Cover End Date does not exceed::',
           textList: ['for a Bond, the Bond Maximum Cover Period; and', 'for a Loan, the Loan Maximum Cover Period'],
         },
         13: {
-          text: 'The Covered Bank Exposure under the Transaction (converted (as at the date this representation is made) for this purpose into the Base Currency) is not more than the lesser of:',
+          text: 'The Covered Bank Exposure under the Transaction (converted (as at the date this representation is made) for this purpose into the Base Currency) is not more than the lesser of::',
```
Comment (lmarrai-sw) [inline trade-finance-manager-api/src/v1/content-strings/index.js:279]: We've accidentally added the `:` twice to these two ECs

**Review by lmarrai-sw — CHANGES_REQUESTED** [review]
Thanks @abhi-markan, just 2 small issues from the find-and-replace

**Review by lmarrai-sw — DISMISSED** [review]
Looks great to me thanks @abhi-markan, just the "fee-s-" question left :)

**Review by lmarrai-sw — APPROVED** [review]

**Pull request merged**